hudi-client/hudi-flink-client/src/main/java/org/apache/hudi/table/action/commit/BaseFlinkCommitActionExecutor.java

```diff
@@ -62,46 +62,54 @@
  * we should avoid that in streaming system.
  */
 public abstract class BaseFlinkCommitActionExecutor<T> extends
-    BaseCommitActionExecutor<T, List<HoodieRecord<T>>, List<HoodieKey>, List<WriteStatus>, HoodieWriteMetadata> {
+    BaseCommitActionExecutor<T, Iterator<HoodieRecord<T>>, List<HoodieKey>, List<WriteStatus>, HoodieWriteMetadata> {
 
   private static final Logger LOG = LoggerFactory.getLogger(BaseFlinkCommitActionExecutor.class);
 
   protected HoodieWriteHandle<?, ?, ?, ?> writeHandle;
 
+  protected final BucketInfo bucketInfo;
+
+  public BaseFlinkCommitActionExecutor(HoodieEngineContext context,
+                                       HoodieWriteHandle<?, ?, ?, ?> writeHandle,
+                                       HoodieWriteConfig config,
+                                       HoodieTable table,
+                                       String instantTime,
+                                       WriteOperationType operationType) {
+    this(context, writeHandle, null, config, table, instantTime, operationType, Option.empty());
```

Comment:
the `bucketInfo` should be never null?

Reply:
Some subclasses do not have bucket info context, e.g., `FlinkPartitionTTLActionExecutor`.